src/negotiation/http/maker.ts

```diff
@@ -0,0 +1,47 @@
+/// An Http implementation of the maker negotiation service
+
+import express from "express";
+import { Maker } from "../maker";
+import { Order } from "../payload";
+
+export class HttpMaker {
+  private readonly maker: Maker;
+
+  constructor(maker: Maker) {
+    this.maker = maker;
+  }
+
+  public listen(port: number) {
+    const app = express();
+
+    app.get("/", (_, res) =>
+      res.send("Maker's Negotiation Service is up and running!")
+    );
+
+    app.get("/orders/:tradingPair", (req, res) => {
+      const order = this.maker.getOrderForTradingPair(req.params.tradingPair);
+      if (!order) {
+        res.status(404).send("Order not found");
+      } else {
+        res.send(order);
+      }
+    });
+
+    app.get("/orders/:keypair/:orderid", (req, res) => {
+      const order = this.maker.getOrderById(req.params.orderid);
+      if (!order || req.params.keypair !== order.tradingPair) {
+        res.status(404).send("Order not found");
+      } else {
+        res.send(this.maker.getExecutionParams());
+      }
+    });
+
+    app.listen(port, () =>
+      console.log(`Maker's Negotiation Service is listening on port ${port}.`)
+    );
+  }
+
+  public addOrder(order: Order) {
```

Comment:
📣
If you have access to the `maker` from the outside who holds all the order then you don't need this function.

Reply:
Not sure I understood this comment so please check again and re-iterate if still not satisfied.